### PR TITLE
Change TenantIdQuery to also query with EventStampName column

### PR DESCRIPTION
**Pull request by hforeste**
Cloud service domain name suffix is different across clouds, instead of coding for multiple domain names eg., `*.cloudapp.net, *.chinacloudapp.cn` and etc, we should use EventStampName instead which is the SourceMoniker equivalent.

### CODE REVIEW

File: src/Diagnostics.RuntimeHost/Services/ResourceService/StampService.cs [hunk 1]
```diff
@@ -104,7 +104,7 @@ private string GetTenantIdQuery(string stamp, DateTime startTime, DateTime endTi
             return
                 $@"{tableName}
                 | where TIMESTAMP >= datetime({startTimeStr}) and TIMESTAMP <= datetime({endTimeStr})
-                | where PublicHost =~ ""{stamp}.cloudapp.net"" or PublicHost matches regex ""{stamp}([a-z{{1}}]).cloudapp.net""
+                | where PublicHost =~ ""{stamp}.cloudapp.net"" or PublicHost matches regex ""{stamp}([a-z{{1}}]).cloudapp.net"" or EventStampName =~ ""{stamp}""
```
Comment (ShekharGupta1988) [inline src/Diagnostics.RuntimeHost/Services/ResourceService/StampService.cs]: Just do a quick check for:

- **Mega stamps** - I think there is another column called 'EventPrimaryStampName' that will hold the mega stamp name and not this column. This 'EventStampName' would have something like 'waws-prod-sn1-107a'
Ex:
```
AntaresIISLogFrontEndTable
| where TIMESTAMP > ago(20m) and EventStampName  == 'waws-prod-sn1-107a'
| summarize by EventStampName, EventPrimaryStampName
```

  |   |  
-- | -- | --
  | **EventStampName** | **EventPrimaryStampName**
  | waws-prod-sn1-107a | waws-prod-sn1-107



- **Sites in ASE** - Make sure you are seeing the right name in this column which is passed in the stamp parameter to this method and there is no friendly name issue.

File: src/Diagnostics.RuntimeHost/Services/ResourceService/NationalCloudStampService.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+            {
+                return await Task.FromResult(new List<string>());
+            }
+        }
+
+        protected override string GetTenantIdQuery(string stamp, string startTimeStr, string endTimeStr, string tableName)
+        {
+            return
+                $@"{RoleInstanceHeartBeatTableName}
+                | where TIMESTAMP >= datetime({startTimeStr}) and TIMESTAMP <= datetime({endTimeStr}) and where EventStampName =~ ""{stamp}""
```
Comment (ShekharGupta1988) [inline src/Diagnostics.RuntimeHost/Services/ResourceService/NationalCloudStampService.cs]: will this work for mega stamp?

Comment (hforeste) [reply src/Diagnostics.RuntimeHost/Services/ResourceService/NationalCloudStampService.cs]: There are no megastamps in China

Comment (ShekharGupta1988) [reply src/Diagnostics.RuntimeHost/Services/ResourceService/NationalCloudStampService.cs]: you should use EventStampPrimary column
in that way, mega stamps are also taken care of..

Comment (hforeste) [reply src/Diagnostics.RuntimeHost/Services/ResourceService/NationalCloudStampService.cs]: So this column actually does not even exist there in China Kusto

Comment (ShekharGupta1988) [reply src/Diagnostics.RuntimeHost/Services/ResourceService/NationalCloudStampService.cs]: ok.. cool. I think for now, its fine then.. Approved the PR

**Pull request merged**